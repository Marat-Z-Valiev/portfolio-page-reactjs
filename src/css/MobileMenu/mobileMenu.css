.hamburger-button {
  background: transparent;
  border: none;
  margin-right: 15px;
  color: #ffffff;
}
.navbar-mobile {
  display: flex;
  background-color: #4594a4;
  font-size: 1.7em;
  height: 100vh;
  color: #ffffff;
}
.navbar-mobile .nav-list-mobile {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  height: 70%;
  padding: 0;
}
.navbar-mobile .nav-list-mobile .nav-list-item {
  display: block;
  margin-right: 20px;
}
.navbar-mobile .nav-list-mobile a {
  text-decoration: none;
}
.navbar-mobile .nav-list-mobile a:hover {
  color: #ffffff;
  cursor: pointer;
}
.opem {
  display: flex;
}
