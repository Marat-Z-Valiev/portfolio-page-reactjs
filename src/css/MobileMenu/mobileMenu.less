.hamburger-button {
	background: transparent;
	border: none;
	margin-right: 15px;
	color: #ffffff;
}

.mobile-menu {
	display: flex;
	background-color: #4594a4;
	font-size: 1.7em;
	height: 100vh;
	color: #ffffff;
	position: fixed;
	transform: translateX(-100%);
	transition: transform 0.5s ease-out;
	z-index: 100;
	width: 100%;

	.nav-list-mobile {
		display: flex;
		flex-direction: column;
		justify-content: space-evenly;
		align-items: center;
		width: 100%;
		height: 70%;
		padding: 0;

		.nav-list-item {
			display: block;
			margin-right: 20px;
		}

		a {
			text-decoration: none;
			&:hover {
				color: #ffffff;
				cursor: pointer;
			}
		}
	}
}
.open {
	transform: translateY(0);
}
