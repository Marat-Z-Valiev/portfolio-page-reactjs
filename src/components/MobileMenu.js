import React from "react";
import HeaderLink from "./HeaderLink";
import headerLinks from "../headerLinks";
import "../css/MobileMenu/mobileMenu.css";

class MobileMenu extends React.Component {
	render() {
		const openClass = this.props.visible ? "open" : "";
		return (
			<nav className={`mobile-menu ${openClass}`}>
				<ul className="nav-list-mobile">
					{headerLinks.map((menuItem, index) => (
						<HeaderLink
							key={index}
							link={menuItem.link}
							label={menuItem.label}
						/>
					))}
				</ul>
			</nav>
		);
	}
}

export default MobileMenu;
