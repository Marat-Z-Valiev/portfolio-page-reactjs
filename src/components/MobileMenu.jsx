import React from "react";
import PropTypes from "prop-types";
import HeaderLink from "./HeaderLink";
import headerLinks from "../headerLinks";
import "../css/MobileMenu/mobileMenu.css";

class MobileMenu extends React.Component {
	closeMenu = () => {
		document.querySelector(".hamburger-button").click();
	};

	render() {
		const openClass = this.props.isMobileMenuVisible ? "open" : "";
		return (
			<nav className={`mobile-menu ${openClass}`}>
				<ul className="nav-list-mobile">
					{headerLinks.map((menuItem, index) => (
						<HeaderLink
							key={index}
							link={menuItem.link}
							label={menuItem.label}
							closeMenu={this.closeMenu}
						/>
					))}
				</ul>
			</nav>
		);
	}
}

MobileMenu.propTypes = {
	visible: PropTypes.bool
};

export default MobileMenu;
