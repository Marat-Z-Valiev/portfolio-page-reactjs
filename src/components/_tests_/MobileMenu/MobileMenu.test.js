import React from "react";
import {shallow} from "enzyme";

import MobileMenu from "../../MobileMenu";

const component = shallow(<MobileMenu />);

describe("MobileMenu", () => {
	it("should render correctly", () => {
		expect(component).toMatchSnapshot();
	});
});
