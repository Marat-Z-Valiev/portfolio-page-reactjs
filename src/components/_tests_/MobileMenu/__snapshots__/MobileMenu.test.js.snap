// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav-list-mobile"
      >
        <HeaderLink
          closeMenu={[Function]}
          label="Home"
          link="home"
        />
        <HeaderLink
          closeMenu={[Function]}
          label="About"
          link="about"
        />
        <HeaderLink
          closeMenu={[Function]}
          label="Projects"
          link="projects"
        />
        <HeaderLink
          closeMenu={[Function]}
          label="Contact"
          link="contact"
        />
      </ul>,
      "className": "mobile-menu ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderLink
            closeMenu={[Function]}
            label="Home"
            link="home"
          />,
          <HeaderLink
            closeMenu={[Function]}
            label="About"
            link="about"
          />,
          <HeaderLink
            closeMenu={[Function]}
            label="Projects"
            link="projects"
          />,
          <HeaderLink
            closeMenu={[Function]}
            label="Contact"
            link="contact"
          />,
        ],
        "className": "nav-list-mobile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "closeMenu": [Function],
            "label": "Home",
            "link": "home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "closeMenu": [Function],
            "label": "About",
            "link": "about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "closeMenu": [Function],
            "label": "Projects",
            "link": "projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "closeMenu": [Function],
            "label": "Contact",
            "link": "contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav-list-mobile"
        >
          <HeaderLink
            closeMenu={[Function]}
            label="Home"
            link="home"
          />
          <HeaderLink
            closeMenu={[Function]}
            label="About"
            link="about"
          />
          <HeaderLink
            closeMenu={[Function]}
            label="Projects"
            link="projects"
          />
          <HeaderLink
            closeMenu={[Function]}
            label="Contact"
            link="contact"
          />
        </ul>,
        "className": "mobile-menu ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HeaderLink
              closeMenu={[Function]}
              label="Home"
              link="home"
            />,
            <HeaderLink
              closeMenu={[Function]}
              label="About"
              link="about"
            />,
            <HeaderLink
              closeMenu={[Function]}
              label="Projects"
              link="projects"
            />,
            <HeaderLink
              closeMenu={[Function]}
              label="Contact"
              link="contact"
            />,
          ],
          "className": "nav-list-mobile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "closeMenu": [Function],
              "label": "Home",
              "link": "home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "closeMenu": [Function],
              "label": "About",
              "link": "about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "closeMenu": [Function],
              "label": "Projects",
              "link": "projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "closeMenu": [Function],
              "label": "Contact",
              "link": "contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
